### PR TITLE
MP-Metrics 3.0 [NoShip]: REST stat metrics to track unmapped exceptions

### PR DESCRIPTION
**Pull request by Channyboy**
#build
#run-libby-bot

### CODE REVIEW

**Comment by Channyboy** [comment]
#build
#run-libby-bot

**Comment by Channyboy** [comment]
#build 
#run-libby-bot

**Comment by Channyboy** [comment]
A timeout occurred in the `mpMetrics-2.3` FAT TCK.
Does not seem to be related. But will run another build.
#build

File: dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package com.ibm.ws.microprofile.metrics.monitor;
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java:1]: Copyright is missing

File: dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		Map.Entry<String, String> classXmethod = resolveSimpleTimerClassMethodTags(resourceInfo);
+
+		registerOrRetrieveRESTUnmappedExceptionMetric(classXmethod.getKey() ,classXmethod.getValue()).inc();
+		
+		return Collections.singletonMap(EXCEPTION_KEY, t);
+	}
+
+	static SharedMetricRegistries sharedMetricRegistry;
+	
+	@Reference
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java:62]: Irregular indentation

File: dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    public void getSharedMetricRegistries(SharedMetricRegistries sharedMetricRegistry) {
+		MetricsJaxRsEMCallbackImpl.sharedMetricRegistry = sharedMetricRegistry;
+    }
+	
+	
+	private Map.Entry<String, String> resolveSimpleTimerClassMethodTags(ResourceInfo resourceInfo) {
+		Class<?> resourceClass = resourceInfo.getResourceClass();
```
Comment (fmhwong) [inline dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java:69]: Do you need a null check on `resourceInfo`?

File: dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java (hunk 1 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java:11]: fixed

File: dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java (hunk 2 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java:72]: fixed

File: dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java (hunk 3 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java:79]: Hm, this is called by the callback handler in response to a JAX-RS request that ended with an unmapped exception. I don't think the `ResourceInfo` would be `null`. @jim-krueger  what do you think?

Comment (jim-krueger) [reply dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java:79]: Correct @Channyboy , resourceInfo should never be null.

**Comment by Channyboy** [comment]
#run-libby-bot

**Pull request merged**